### PR TITLE
fix compiler crash

### LINKED ISSUE #25663: 'const enum' in single statement context crashes compiler
**TypeScript Version:**  3.0.0-dev.20180712

**Search Terms:** 

**Code**

```ts
if (Boolean())
  const enum Foo {}
```

**Expected behavior:**

Some compiler error.

**Actual behavior:**

```
TypeError: Cannot read property 'kind' of undefined
    at Object.isBlock (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:9915:21)
    at emitEmbeddedStatement (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:65524:20)
    at emitIfStatement (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:64709:13)
    at pipelineEmitWithHint (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:63854:32)
    at emitNodeWithSourceMap (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:62836:21)
    at pipelineEmitWithSourceMap (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:63726:58)
    at emitNodeWithComments (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:62970:17)
    at pipelineEmitWithComments (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:63718:13)
    at emitNodeWithNotification (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:62486:21)
    at pipelineEmitWithNotification (/Users/klaus/code/wotan/node_modules/typescript/lib/tsc.js:63713:47)
```

**Playground Link:** https://agentcooper.github.io/typescript-play/#code/JYMwBAFAQg9jA2BTAhgOwgSgwKDHsAxjKgM4AuYiqArgLZgBicYA3gL5A

**Related Issues:**

#22682 - about declarations in single statement context
#22686 - similar error for interfaces and type aliases

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #25663

### CODE REVIEW

File: tests/cases/conformance/constEnums/constEnum4.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if (true) {
```
Comment (ajafff) [inline tests/cases/conformance/constEnums/constEnum4.ts]: You need to remove the curly braces to actually test your change

Comment (Kingwl) [reply tests/cases/conformance/constEnums/constEnum4.ts]: sorry my fault

**Comment by RyanCavanaugh** [comment]
@rbuckton please merge if this looks correct - thanks

**Comment by Kingwl** [comment]
(up)

**Pull request merged**